digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -223,6 +225,57 @@ class RoadAddressService(roadLinkService: RoadLinkService, roadwayDAO: RoadwayDA
     }
   }
 
+  def collectResult(dataType: String, inputData: Seq[Any], oldSeq: Seq[Map[String, Seq[Any]]]): Seq[Map[String, Seq[Any]]] = {
+    val result: Seq[Map[String, Seq[Any]]] = Seq(Map((dataType, inputData)))
+    val newSeq: Seq[Map[String, Seq[Any]]] = oldSeq ++ result
+    return newSeq
+  }
+
+  def getSearchResults(searchString: Option[String]): Seq[Map[String, Seq[Any]]] = {
+    logger.debug("getSearchResults")
+    val parsedInput = locationInputParser(searchString)
+    val searchType = parsedInput.head._1
+    var searchResult: Seq[Any] = null
+    var resultSeq: Seq[Map[String, Seq[Any]]] = Seq.empty[Map[String, Seq[Any]]]
+    if (searchType == "road") {
+      val nums = parsedInput.head._2
+      if (nums.size == 3) {
+        searchResult = getRoadAddress(nums(0), nums(1), nums(2), None).sortBy(address => (address.roadPartNumber, address.startAddrMValue))
+        resultSeq = collectResult("road", searchResult, resultSeq)
+      } else if (nums.size == 2) {
+        searchResult = getRoadAddressWithRoadNumberParts(nums(0), Set(nums(1)), Set(Track.Combined, Track.LeftSide, Track.RightSide)).sortBy(address => (address.roadPartNumber, address.startAddrMValue))
+        resultSeq = collectResult("road", searchResult, resultSeq)
+      } else if (nums.size == 1) {
```

Comment:
should we deal with the else case option where num.size can be bigger than 3?

Reply:
bigger values are now handled by returning resultSeq with inital value empty Sequence  as search result